spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair_spec.rb

```diff
@@ -0,0 +1,43 @@
+describe ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::AuthKeyPair do
+  let(:ems) do
+    FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud, :provider_region => "us-south")
+  end
+
+  let(:the_raw_key_pair) do
+    double.tap do |key_pair|
+      allow(key_pair).to receive(:name).and_return('key1')
+      allow(key_pair).to receive(:fingerprint).and_return('0000')
+      allow(key_pair).to receive(:public_key).and_return('AAA...B')
+      allow(key_pair).to receive(:private_key).and_return('BBB...C')
+    end
+  end
+
+  describe 'key pair create and delete' do
+    it 'creates new key pair in nova' do
+      service = double
+      key_pairs = double
+      allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems)
+      allow(ems).to receive(:connect).with(:service => 'Compute').and_return(service)
```

Comment:
We need `:service => 'PowerIaas'` [see ManagerMixin#connect](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/manager_mixin.rb#L12-L24)

Reply:
This still fails, but I think it's actually a bug. We're calling `connect(:target => "PowerIaas")` instead of `:service` in two places:
https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair.rb#L5
https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair.rb#L25

@agrare am I right that this is this wrong?

Reply:
See #36

Reply:
Yes you're right @jaywcarman that was only working because `PowerIaas` is the default (`options[:service] ||= "PowerIaas"`)